File: src/main/java/com/github/kdm1jkm/baseballSolver/condition/SingleCondition/OutCondition.java
package com.github.kdm1jkm.baseballSolver.condition.SingleCondition;

public class OutCondition implements SingleCondition{
    private final int num;

    public OutCondition(int num) {
        this.num = num;
    }


    @Override
    public boolean isTrue(int[] nums) {
        for(int n : nums){
            if(n ==num )return false;
        }
        return true;
    }

    @Override
    public int getNum() {
        return num;
    }

    @Override
    public SingleCondition copy() {
        return new OutCondition(num);
    }
}

File: src/main/java/com/github/kdm1jkm/baseballSolver/condition/SingleCondition/OutCondition.kt
package com.github.kdm1jkm.baseballSolver.condition.SingleCondition

import com.github.kdm1jkm.baseballSolver.condition.ConditionSort

class OutCondition(override val num: Int) : SingleCondition, Cloneable {

    override val pos: Int
        get() = -1

    override fun isTrue(nums: IntArray): Boolean {
        for (num in nums) if (num == this.num) return false
        return true
    }

    override val sort: ConditionSort
        get() = ConditionSort.Out

    @Throws(CloneNotSupportedException::class)
    public override fun clone(): Any {
        return super<SingleCondition>.clone()
    }

}